### DISCUSSION TITLE
Please clarify some questions I have about nvidia proprietary vs nvidia-open compatibility with wayland compositors.

### DISCUSSION THREAD

**Discussion by pillowtrucker** [NVIDIA/open-gpu-kernel-modules#566] [Q&A]
1. Why is it such a pervasive myth that one should need nvidia-open drivers for wayland compatibility ? Is it a myth ? I'm still not sure. Have people been confusing nvidia-open with nouveau ?
2. Or it just because of GBM vs EGLS support ? Do the two driver versions differ in that respect ?
3. Does this mean any mir-based (as opposed to wlroots-based) compositor would have worked the entire time, just because they actually put in effort to have an EGLS rendering backend ?
4. Does this (very helpful) list of known issues apply to the closed-source-, the open- driver, or both ? https://download.nvidia.com/XFree86/Linux-x86_64/535.113.01/README/wayland-issues.html

**Comment by pillowtrucker** [comment]
[ xorg/x11 Protocol specification to accompany providing explicit GPU synchronization capabilities to the DRI3 and Present extensions.](https://gitlab.freedesktop.org/xorg/proto/xorgproto/-/merge_requests/59)
[ the implementation thereof ](https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/967)
[ the melodrama thread seemingly about everything that is still broken in wayland and xwayland for some reason contained in a xorg glamor issue](https://gitlab.freedesktop.org/xorg/xserver/-/issues/1317)
[wayland protocol for (an) explicit sync mechanism](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/90)

**Comment by pillowtrucker** [comment]
I'm no longer convinced that GBM/EGLStream compatibility is actually the most important remaining issue, as it seems it's been mostly resolved in the last year by all parties (i.e. NVIDIA, XWayland, wayland compositors and their underlying libraries).
There are several headings under the [nvidia readme](https://download.nvidia.com/XFree86/Linux-x86_64/535.113.01/README/) other than the `wayland-issues` section that contain some additional information on *that* whole situation.

However, after some additional digging through many meandering, often hostile, sometimes childish comments in a few threads on the freedesktop gitlab instance, I think I figured out at least another ingredient of the wayland quagmire: It seems that the main problem remaining is support for explicit synchronization of the gfx drivers, the compositors and client applications.
NVIDIA has added /some/ support for implicit sync in the last year, working around at least some of the problems (although I'm still not sure if that's the open source drivers or the proprietary ones or both). Meanwhile, the explicit sync PR in XWayland is still unmerged and, while some of compositors have support for the protocol, hardly any clients do (XWayland being one of those clients).
@erik-kz @cubanismo if you don't mind, would you please confirm if that is an accurate summary of the situation ?

Maybe if I spend another few weeks trying to wrap my head around all this I'll figure out why/how/when/if PRIME offloading works (under wayland) or if it's ever going to work (on my fragile laptop that already requires some funny software from https://asus-linux.org/ for it to function at all.)

**Reply by aritger** [reply]
I'm not aware of anything for Wayland that specifically requires open-gpu-kernel-modules versus the NVIDIA proprietary kernel modules.  The Wayland support with the NVIDIA GPU driver stack should be the same regardless of which flavor of kernel modules you use.

Yes, there is ongoing work to continue to improve our Wayland support.  It looks like you already found links to where some of those code reviews are happening.
